packages/good-design/src/core/web3/modals/BasicStyledModal.tsx

```diff
@@ -56,17 +57,50 @@ export const ModalLoaderBody = () => (
 export const ModalErrorBody = ({ error }: { error: string }) => (
   <VStack space={6} justifyContent="center" alignItems="center">
     <Image source={BillyOops} w={137} h={135} style={{ resizeMode: "contain" }} />
-    <Text variant="sm-grey" color="goodRed.100">
+    <Text variant="sm-grey-650" color="goodRed.100">
       {error}
     </Text>
   </VStack>
 );
 
-export const ModalFooterCta = ({ buttonText, action }: { buttonText: string; action: () => void }) => (
-  <Center padding="0" w="100%">
-    <LinkButton buttonText={buttonText} onPress={action} />
-  </Center>
-);
+export const ModalFooterCta = ({
+  buttonText,
+  withDontShowAgain,
+  action,
+  styleProps
+}: {
+  buttonText: string;
+  withDontShowAgain?: string | undefined;
+  styleProps?: any;
+  action: () => void;
+}) => {
+  const [dontShowAgain, setDontShowAgain] = useState(false);
+
+  const onAction = async () => {
+    if (withDontShowAgain && dontShowAgain) {
+      await AsyncStorage.setItem(withDontShowAgain, "true");
```

Comment:
the key seems incorrect
also where is this actually checked to not show again?

Reply:
1. clarified the expected flow a bit by using different naming
2. I initially thought to handle the key outside of the ui-components, its now handled by the modal
3. small fyi: its a generic modal and the 'do not show again' is not actively used for the good-id details flow

commit: https://github.com/GoodDollar/GoodWeb3-Mono/pull/169/commits/d00d20df045936ffa8b5c5f650c86d7cc8d569ee